docker/Dockerfile

```diff
@@ -3,6 +3,11 @@ ARG build=local
 # Environment
 FROM node:14.17-alpine3.14 AS environment
 
+# Download solidity compiler as per: https://github.com/nomiclabs/hardhat/issues/1280#issuecomment-949822371
+# The binary filename is taken from: https://github.com/ethereum/solc-bin/tree/gh-pages/linux-amd64
+RUN mkdir -p /root/.cache/hardhat-nodejs/compilers/linux-amd64 \
+  &&  wget -O /root/.cache/hardhat-nodejs/compilers/linux-amd64/solc-linux-amd64-v0.8.9+commit.e5eed63a https://binaries.soliditylang.org/linux-amd64/solc-linux-amd64-v0.8.9+commit.e5eed63a
```

Comment:
Why are you running this in the `environment` phase? All other steps inherit from this one, is it needed everywhere? I think the `preparation` phase would be sufficient, wouldn't it?

Reply:
Yeah, it only needs to be downloaded before `yarn install` - thanks for noticing